src/sentry/static/sentry/app/views/alerts/details/index.tsx

```diff
@@ -60,13 +60,18 @@ class IncidentDetails extends React.Component<Props, State> {
     } = this.props;
 
     try {
-      const [incident, stats] = await Promise.all([
-        fetchIncident(api, orgId, alertId),
-        fetchIncidentStats(api, orgId, alertId),
-      ]);
-
-      this.setState({incident, stats, isLoading: false, hasError: false});
-      markIncidentAsSeen(api, orgId, incident);
+      const incidentPromise = fetchIncident(api, orgId, alertId).then(incident => {
+        this.setState({incident});
+        markIncidentAsSeen(api, orgId, incident);
+      });
```

Comment:
Could be convinced of other ways of writing this

Reply:
iirc I initially had it like this but didn't like mixing async/await and promises and thought "how slow can it really be" -- in any case this is ok, the only other alternative I can think of is to have 2 async methods

Reply:
[incident](https://sentry.io/organizations/sentry/performance/summary/?display=durationpercentile&project=1&query=title%3A%2Fapi%2F0%2Forganizations%2F%7Borganization_slug%7D%2Fincidents%2F%7Bincident_identifier%7D%2F+event.type%3Atransaction+project%3Asentry&statsPeriod=30d&transaction=%2Fapi%2F0%2Forganizations%2F%7Borganization_slug%7D%2Fincidents%2F%7Bincident_identifier%7D%2F) vs [stats](https://sentry.io/organizations/sentry/performance/summary/?display=durationpercentile&project=1&query=title%3A%2Fapi%2F0%2Forganizations%2F%7Borganization_slug%7D%2Fincidents%2F%7Bincident_identifier%7D%2Fstats%2F+event.type%3Atransaction+project%3Asentry&statsPeriod=30d&transaction=%2Fapi%2F0%2Forganizations%2F%7Borganization_slug%7D%2Fincidents%2F%7Bincident_identifier%7D%2Fstats%2F) i think the problem with some of the stats is that they hit near 2 seconds and there's still nothing on the page.

Reply:
Didn't mean to dispute the need for this, it was more like a "sorry" for making you fix this.